src/solvers/smt/smt_conv.cpp

```diff
@@ -2297,6 +2327,32 @@ smt_astt smt_convt::array_create(const expr2tc &expr)
   return newsym_ast;
 }
 
+smt_astt smt_convt::array_create(const constant_vector2tc &expr)
+{
+  const vector_type2t &arr_type = to_vector_type(expr->type);
+  return array_create(
+    expr,
+    arr_type.size_is_infinite,
+    arr_type.vector_size,
+    expr->datatype_members);
+}
+
+smt_astt smt_convt::array_create(const expr2tc &expr)
```

Comment:
Add some proper comments here. Just `check size` is not enough.

Reply:
The comment and implementation comes from the original implementation, I've only generalized to work with vectors as well